README.md
# Univention Corporate Client

Welcome to the git mirror of Univention Corporate Client (UCC).

## Documentation and Support

The UCC documentation, including a Quickstart guide, can be found at [docs.software-univention.de](https://docs.software-univention.de/).

If you need direct help, the forum [Univention
Help](https://help.univention.com) provides a very good community support. For commercial support, please have a look at our [support
offerings](https://www.univention.com/download-and-support/support/commercial-support/).

## Contributing

Please read the [contributing guide](./CONTRIBUTING.md) to find more information about the UCS development process, how to propose bugfixes and improvements.
The [Code of Conduct contains guidelines](./CONTRIBUTING.md#code-of-conduct) we expect project participants to adhere to.

## License

Univention Corporate Client is built on top of many existing open source projects which use their own licenses.
The source code of all parts written by Univention like the management system is licensed under the AGPLv3 if not stated otherwise directly in the source code.
Please see the [license file](./LICENSE) for more information.


cdpinger/ChangeLog
  157 Vagrant Cascadian 2012-02-24
      Add symlinks for Debian/Ubuntu for ltspfsd.rules creation in
      init-ltsp.d.

  156 Vagrant Cascadian 2012-02-24
      Use init-ltsp.d hook for installing udev rules at boot.

  155 Vagrant Cascadian 2012-02-24
      Rename udev rules to ltspfsd.rules, and move them into
      /usr/share/ltspfs/udev, which allows for installing rules at boot.

  154 vagrant@freegeek.org	2011-10-28 {ltspfs-1.0}
      version 1.0

  153 vagrant@freegeek.org	2011-10-28 [merge]
      merge patch from Scott Balneaves to fix 64 bit issues.

  152 Stéphane Graber	2011-08-11 {ltspfs-0.9}
      version 0.9

  151 Stéphane Graber	2011-07-22
      Launchpad: no change commit to trigger a test rebuild

  150 Stéphane Graber	2011-07-07
      ID_FS_LABEL is now usually 'safe'. Patch from gcc (LP: #690969)

  149 Warren Togami	2011-06-16 {ltspfs-0.8}
      version 0.8

  148 Vagrant Cascadian 2011-02-05
      cdpinger: add support for UDF formatted CDs and DVDs.

  147 Stéphane Graber	2010-08-08 {ltspfs-0.7}
      Release 0.7

  146 Vagrant Cascadian 2010-06-25
      init with "ltspfs" rather than simply "test".

  145 Vagrant Cascadian 2010-06-25
      Add example ltspfs mounted hook to issue notifications for
      desktops that do not display icons. Thanks to Petter Reinholdtsen.
      http://bugs.debian.org/575031

  144 Gideon Romm	2010-05-11
      Remove commented-out code.

  143 Gideon Romm	2010-05-11
      Fix cdrom handling.

  142 Stéphane Graber	2010-02-17 {ltspfs-0.6}
      version 0.6

  141 Stéphane Graber	2010-02-17
      Try umounting with -l to avoid some issues

  140 Scott Balneaves	2010-02-17
      Change owner of dir in /media to user, with group of root

  139 Scott Balneaves	2010-02-17
      Top level media directory being created mode 750.  Changed to 700

  138 Gideon Romm	2010-02-10
      Be careful to actually have a pid before executing kill

  137 Gideon Romm	2010-02-10
      Do not sed -i a file that we are reading in from stdin

  136 Vagrant Cascadian 2010-02-10
      allow CDs to have an empty filesystem, as they're likely to have
      a device but no disk inserted. this allows cdpinger to run again.

  135 Vagrant Cascadian 2010-02-10
      cdpinger: report syslog messages for "disk detected" and "tray
      open" messages only when the state changes so they are not reported
      every 3 seconds.

  134 Gideon Romm	2010-02-10
      Fix MODE setting to include add_disc annd remove_disc

  133 Gideon Romm	2010-02-10
      Pass DEVICENAME to add_cdrom_device and remove_cdrom_device
      functions

  132 Gideon Romm	2010-02-10
      Make sure ltspfsd gets launched

  131 Gideon Romm	2010-02-10
      Refactor ltspfs_entry and cdpinger code to better handle CDROMs

  130 Gideon Romm	2010-02-10
      Remove cdpinger pid file when cdrom is removed, since apparently
      cdpinger will commit suicide on its own upon device removal and
      we would like cdpinger to start up again upon reinsertion of the
      cdrom drive.

  129 Gideon Romm	2010-02-09
      Try to avoid multiple parallel calls to pgrep by writing a pid
      file to check first. If the pid file exists, it shell should not
      need to use pgrep in the conditional

  128 Gideon Romm	2010-02-01
      Make delayed-mounter work for LDM localapps

  127 Scott Balneaves	2010-01-30
      Added syslogging.

  126 Gideon Romm	2010-01-29
       Some more optimizations to use shell primitives where possible

  125 Vagrant Cascadian 2010-01-29
      fix call_cdpinger function to use exact matching so that "cdrom"
      doesn't match "cdrom1". thanks gadi!

  124 Gideon Romm	2010-01-27
      Fix up noauth stuff and make local devices work for localapps,
      rdesktop, and kiosk scripts.

  123 Scott Balneaves	2010-01-27
      Changed noauth mode: ltspfs now recognizes LTSPFS_NOAUTH env var,
      and if set, skips auth.  ltspfsd now skips receiving auth packet
      if noauth set.

  122 Stéphane Graber	2010-01-13 {ltspfs-0.5.14}
      release ltspfs 0.5.14

  121 Stéphane Graber	2010-01-13
      Fix cases where the user used for login doesn't match the case of
      the remote user

  120 Scott Balneaves	2010-01-12
      Files are now user-owned only

  119 Marc Gariepy	2009-12-15
      fixing local lowercase username (ldap ltsp-trunk -r 1265)

  118 Vagrant Cascadian 2009-07-17 {ltspfs-0.5.13}
      release ltspfs 0.5.13

  117 Vagrant Cascadian 2009-07-17
      drop delayedmounter from xinitrc.d, as udev scripts should be
      sufficient now.

  116 Vagrant Cascadian 2009-07-06
      relocate delayed-mounter ldm hooks so they run after the user
      is added to /etc/passwd, and before the user is removed from
      /etc/passwd.

  115 Vagrant Cascadian 2009-07-06
      - always run as root, so mounts appear in /media/root - only run
      mount for localapps if user's home dir is mounted via sshfs

  114 Vagrant Cascadian 2009-07-06
      check for ldm-xauth in /var/run, and support newer ldm, with
      Xauthority in a sub-dir.

  113 Vagrant Cascadian 2009-06-21
      fix spelling of "Received" in ltspfsd

  112 Stéphane Graber	2009-06-03 {ltspfs-0.5.12}
      version 0.5.12

  111 Francis Giraldeau 2009-05-27
      remove ltspfs_fstab entry before unmounting device,
      to prevent other processes from mounting it.
      https://bugs.launchpad.net/ubuntu/+source/ltspfsd/+bug/378495

  110 Vagrant Cascadian 2009-04-03 {ltspfs-0.5.11}
      version 0.5.11

  109 Vagrant Cascadian 2009-04-03
      simplify code to handle when mountpoint is already mounted

  108 Gideon Romm	2009-03-11
      Make sure delayed_mounter sets USER=root for local ltspfsmounting

  107 Gideon Romm	2009-03-10
      Do not remove ltspfs mount on "already mounted" errors, as these
      errors may occur silently with multiple ltspfsmounters

  106 Stéphane Graber	2009-03-09 {ltspfs-0.5.10}
      version 0.5.10

  105 Stéphane Graber	2009-03-08

cdpinger/src/ltspfs.h
/*
 * Defines
 */

/*
 * The maximum sized command we have is the symlink command.
 * We'll need: 1 BYTES_PER_XDR_UNIT for the packet length.
 *             1 BYTES_PER_XDR_UNIT for the packet type
 *             2 * (BYTES_PER_XDR_UNIT + PATH_MAX) for the paths.
 * So:
 *
 * ((4 * BYTES_PER_XDR_UNIT) + (2 * PATH_MAX))
 */

#define SERVER_PORT        9220
#define LTSP_MAXBUF        ((4 * BYTES_PER_XDR_UNIT) + (2 * PATH_MAX))
#define LTSPFS_TIMEOUT     120
#define AUTOMOUNT_TIMEOUT  5
#define LTSP_STATUS_OK     0
#define LTSP_STATUS_FAIL   1
#define LTSP_STATUS_CONT   2
#define PING_INTERVAL      30

/*
 * Packet types
 */

#define LTSPFS_GETATTR     0
#define LTSPFS_READLINK    1
#define LTSPFS_READDIR     2
#define LTSPFS_MKNOD       3
#define LTSPFS_MKDIR       4
#define LTSPFS_SYMLINK     5
#define LTSPFS_UNLINK      6
#define LTSPFS_RMDIR       7
#define LTSPFS_RENAME      8
#define LTSPFS_LINK        9
#define LTSPFS_CHMOD       10
#define LTSPFS_CHOWN       11
#define LTSPFS_TRUNCATE    12
#define LTSPFS_UTIME       13
#define LTSPFS_OPEN        14
#define LTSPFS_READ        15
#define LTSPFS_WRITE       16
#define LTSPFS_STATFS      17
#define LTSPFS_RELEASE     18		/* Not currently used */
#define LTSPFS_RSYNC       19		/* Not currently used */
#define LTSPFS_SETXATTR    20		/* Not currently used */
#define LTSPFS_GETXATTR    21		/* Not currently used */
#define LTSPFS_LISTXATTR   22		/* Not currently used */
#define LTSPFS_REMOVEXATTR 23		/* Not currently used */
#define LTSPFS_XAUTH       24
#define LTSPFS_MOUNT       25
#define LTSPFS_PING        26
#define LTSPFS_QUIT        27

